include/extractor/node_based_graph_factory.hpp

```diff
@@ -36,12 +36,13 @@ class NodeBasedGraphFactory
     // node-based graph to represent the OSM network. This includes geometry compression, annotation
     // data optimisation and many other aspects. After this step, the edge-based graph factory can
     // turn the graph into the routing graph to be used with the navigation algorithms.
-    NodeBasedGraphFactory(const boost::filesystem::path &input_file,
-                          ScriptingEnvironment &scripting_environment,
-                          std::vector<TurnRestriction> &turn_restrictions,
-                          std::vector<ConditionalTurnRestriction> &conditional_turn_restrictions);
+    NodeBasedGraphFactory(const boost::filesystem::path &input_file);
 
-    auto const &GetGraph() const { return compressed_output_graph; }
+    void CompressAll(ScriptingEnvironment &scripting_environment,
```

Comment:
Any documentation on what this function does would be great, as we can now (as it seems) generate an incomplete state and need to compress the graph before we can process it in the edge-based graph factory.
I would like to avoid settings like these.

Is there a reason we cannot operate on the compressed graph? Any edge that will be compressed will not be segregated anyhow.

Reply:
Ok, will introduce segregated processing for compressed node-based graph as discussed.